### PR TITLE
test(ui): Convert group similar issues to RTL

### PR DESCRIPTION
**Pull request by scttcper**

### CODE REVIEW

File: tests/js/spec/views/organizationGroupDetails/groupSimilarIssues.spec.jsx [hunk 1]
```diff
@@ -41,77 +44,69 @@ describe('Issues Similar View', function () {
...
         project={project}
         params={{orgId: 'org-slug', groupId: 'group-id'}}
         location={{}}
       />,
-      routerContext
+      {context: routerContext}
     );
 
-    expect(component).toSnapshot();
+    expect(screen.getByTestId('loading-indicator')).toBeInTheDocument();
```
Comment (scttcper) [inline tests/js/spec/views/organizationGroupDetails/groupSimilarIssues.spec.jsx:62]: removing this snapshot test

File: tests/js/spec/views/organizationGroupDetails/groupSimilarIssues.spec.jsx [hunk 2]
```diff
@@ -41,77 +44,69 @@ describe('Issues Similar View', function () {
...
+        params={{orgId: 'org-slug', projectId: 'project-slug', groupId: 'group-id'}}
+        location={{}}
+      />,
+      {context: routerContext}
     );
+    renderGlobalModal();
+
+    userEvent.click(await screen.findByTestId('similar-item-row'));
+    userEvent.click(await screen.findByRole('button', {name: 'Merge (1)'}));
+    userEvent.click(await screen.findByRole('button', {name: 'Confirm'}));
```
Comment (priscilawebdev) [inline tests/js/spec/views/organizationGroupDetails/groupSimilarIssues.spec.jsx:100]: do we need to use `await` here too?

Comment (scttcper) [reply tests/js/spec/views/organizationGroupDetails/groupSimilarIssues.spec.jsx:100]: yeah it's a lot of awaiting, the confirm button here is coming from a modal

Comment (priscilawebdev) [reply tests/js/spec/views/organizationGroupDetails/groupSimilarIssues.spec.jsx:100]: so we need them? can we maybe just wait for the last button, the confirm?

Comment (scttcper) [reply tests/js/spec/views/organizationGroupDetails/groupSimilarIssues.spec.jsx]: got rid of one of them!

File: tests/js/spec/views/organizationGroupDetails/groupSimilarIssues.spec.jsx [hunk 3]
```diff
@@ -41,77 +44,69 @@ describe('Issues Similar View', function () {
...
-    expect(merge).toHaveBeenCalledWith(
-      '/projects/org-slug/project-slug/issues/',
-      expect.objectContaining({
-        data: {merge: 1},
-      })
+    render(
+      <GroupSimilarIssues
+        project={project}
+        params={{orgId: 'org-slug', projectId: 'project-slug', groupId: 'group-id'}}
+        location={{}}
```
Comment (priscilawebdev) [inline tests/js/spec/views/organizationGroupDetails/groupSimilarIssues.spec.jsx:92]:
```suggestion
   project={project}
        params={{orgId: 'org-slug', groupId: 'group-id'}}
        location={router.location}
        router={router}
        routeParams={router.params}
        routes={router.routes}
        route={{}}
```

File: tests/js/spec/views/organizationGroupDetails/groupSimilarIssues.spec.jsx [hunk 4]
```diff
@@ -41,77 +44,69 @@ describe('Issues Similar View', function () {
...
+    expect(screen.getByTestId('loading-indicator')).toBeInTheDocument();
   });
 
   it('renders with mocked data', async function () {
-    const wrapper = mountWithTheme(
+    const wrapper = render(
       <GroupSimilarIssues
         project={project}
         query=""
         params={{orgId: 'org-slug', projectId: 'project-slug', groupId: 'group-id'}}
```
Comment (priscilawebdev) [inline tests/js/spec/views/organizationGroupDetails/groupSimilarIssues.spec.jsx:70]: should we remove `projectId` from the params?

**Review by priscilawebdev — APPROVED** [review]
left some suggestions but it looks good to me

File: tests/js/spec/views/organizationGroupDetails/groupSimilarIssues.spec.jsx (hunk 3 above)
Comment (priscilawebdev) [reply tests/js/spec/views/organizationGroupDetails/groupSimilarIssues.spec.jsx:92]: for these suggested changes is necessary to have `const router = TestStubs.router();`

**Pull request merged**